### PR TITLE
Add new `Sorbet/ImplicitConversionMethod` cop

### PR DESCRIPTION
**Pull request by dduugg**
Resolves https://github.com/Shopify/rubocop-sorbet/issues/163

Adds a cop to forbid implicit conversion methods. For conciseness, the cop name is consistent with rubocop core's style of naming the thing that is disallowed, (e.g. `DuplicateMethod`), rather than the more verbose `Forbid*` style (e.g. `ForbidTUntyped`).

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ task :new_cop, [:cop] do |_task, args|
   require "rubocop"
 
   cop_name = args.fetch(:cop) do
-    warn("usage: bundle exec rake new_cop[Department/Name]")
+    warn('usage: bundle exec rake "new_cop[Department/Name]"')
```
Comment (Morriar) [inline Rakefile:21]: Just in case we need to revert the current change, do you mind moving this to a separate PR please 🙏

File: lib/rubocop/cop/sorbet/implicit_conversion_method.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      #   def self.to_str(x); end
+      #
+      #   # bad
+      #   alias to_str to_s
+      #
+      # @see https://docs.ruby-lang.org/en/master/implicit_conversion_rdoc.html
+      # @note Since the arity of aliased methods is not checked, false positives may result.
+      class ImplicitConversionMethod < RuboCop::Cop::Base
+        IMPLICIT_CONVERSION_METHODS = [:to_ary, :to_int, :to_hash, :to_str].freeze
+        MSG = "Avoid implicit conversion methods, as Sorbet does not support them."
```
Comment (Morriar) [inline lib/rubocop/cop/sorbet/implicit_conversion_method.rb:33]: Should we suggest an alternative?

**Comment by dduugg** [comment]
@Morriar I believe I've addressed the initial PR comments, PTAL

File: lib/rubocop/cop/sorbet/implicit_conversion_method.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        def on_def(node)
+          return unless IMPLICIT_CONVERSION_METHODS.include?(node.method_name)
+          return unless node.arguments.empty?
+
+          add_offense(node)
+        end
+        alias_method :on_defs, :on_def
+
+        def on_send(node)
+          add_offense(node.first_argument) if IMPLICIT_CONVERSION_METHODS.include?(node.first_argument.value)
```
Comment (Morriar) [inline lib/rubocop/cop/sorbet/implicit_conversion_method.rb:51]: This is going to add an offense for call as well right? Should we add this to the documentation and also add a test for it?

File: lib/rubocop/cop/sorbet/implicit_conversion_method.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          add_offense(new_id) if IMPLICIT_CONVERSION_METHODS.include?(new_id.value)
+        end
+
+        def on_def(node)
+          return unless IMPLICIT_CONVERSION_METHODS.include?(node.method_name)
+          return unless node.arguments.empty?
+
+          add_offense(node)
+        end
+        alias_method :on_defs, :on_def
```
Comment (Morriar) [inline lib/rubocop/cop/sorbet/implicit_conversion_method.rb:48]: Do you mind adding a test for this. I see it's already properly documented.

**Review by Morriar — COMMENTED** [review]
Thanks a lot!

Just two details and it should be good to go 👍
